client/src/components/Categories.jsx

```diff
@@ -1,8 +1,111 @@
 import React, { Component } from 'react';
-
+import IncomeCategoriesDataService from '../services/income-categories.service';
 
 export default class Categories extends Component {
+  constructor(props) {
+    super(props);
+    this.onChangeName = this.onChangeName.bind(this);
+    this.onChangeDescription = this.onChangeDescription.bind(this);
+    this.saveCategory = this.saveCategory.bind(this);
+    this.newCategory = this.newCategory.bind(this);
+
+    this.state = {
+      id: null,
+      name: '',
+      description: '',
+
+      submitted: false,
+    };
+  }
+
+  onChangeName(e) {
+    this.setState({
+      name: e.target.value,
+    });
+  }
+
+  onChangeDescription(e) {
+    this.setState({
+      description: e.target.value,
+    });
+  }
+
+  saveCategory() {
+    const data = {
+      name: this.state.name,
+      description: this.state.description,
+    };
+
+    IncomeCategoriesDataService.create(data)
+      .then((response) => {
+        this.setState({
+          id: response.data.id,
+          name: response.data.name,
+          description: response.data.description,
+
+          submitted: true,
+        });
+        console.log(response.data);
```

Comment:
i would remove this console.log, its bad habit leave console.logs in source code